src/number_format.js

```diff
@@ -128,12 +129,14 @@ class NumberFormat extends React.Component {
   getFloatString(num, props) {
     props = props || this.props;
     const {decimalSeparator, thousandSeparator} = this.getSeparators(props);
-    return num.replace(new RegExp(escapeRegExp(thousandSeparator || ''), 'g'), '').replace(decimalSeparator, '.');
+    const number = num || '';
```

Comment:
No no, it will not override `null`. Only if you pass `undefined` as the first argument, num will be `''`